src/metrics.cc

```diff
@@ -378,6 +412,134 @@ Metrics::PollCacheMetrics(std::shared_ptr<RequestResponseCache> response_cache)
   return true;
 }
 
+#ifdef TRITON_ENABLE_METRICS_CPU
+bool
+Metrics::ParseCpuInfo(std::shared_ptr<CpuInfo> info)
+{
+#ifdef _WIN32
+  // Note: CPU metrics not supported on Windows
+  return false;
+#else
+  std::ifstream ifs("/proc/stat");
+  if (!ifs.good()) {
+    return false;
+  }
+
+  std::string line;
+  std::getline(ifs, line);
+  // Verify first line is aggregate cpu line
+  if (line.rfind("cpu ", 0) == std::string::npos) {
+    return false;
+  }
+
+  std::string _;
+  std::istringstream iss(line);
+  // Use _ to skip "cpu" at start of line
+  if (!(iss >> _ >> info->user >> info->nice >> info->system >> info->idle)) {
+    return false;  // Parsing error
+  }
+  return true;
+#endif  // OS
+}
+
+bool
+Metrics::ParseMemInfo(std::shared_ptr<MemInfo> info)
+{
+#ifdef _WIN32
+  // Note: CPU metrics not supported on Windows
+  return false;
+#else
+  std::ifstream ifs("/proc/meminfo");
+  if (!ifs.good()) {
+    return false;
+  }
+
+  std::string line;
+  constexpr uint64_t KB = 1024;
+  while (std::getline(ifs, line)) {
+    std::istringstream iss(line);
+    std::string name;
+    uint64_t value = 0;
+    if (iss >> name >> value) {
+      // Remove trailing ":" after field names, convert KB to bytes
+      name.pop_back();
+      (*info)[name] = value * KB;
```

Comment:
I compared this code with only  parsing for the "MemTotal" and "MemAvailable" fields and exiting early once they're found, and found minimal difference in performance. 

For 1000 iterations of each:
```
exiting early after finding both fields:   avg=3.454ms   // + extra code to track when we're done
parsing all fields:                        avg=3.524ms
```